### PR TITLE
Fixed defective search

### PR DESCRIPTION
**Pull request by Shagequi**
- Fixed onbackpressed bug
- Fixed the ever-present cross
- Optimized/cleaned the code
~- Added translations (EN, FR, IT, ES, DE) for next step : search by direct edit~

### CODE REVIEW

File: src/main/res/values-fr/strings.xml [hunk 1]
```diff
@@ -307,7 +307,7 @@ Attention la suppression est irréversible.</string>
     <string name="file_migration_cleaning">Nettoyage…</string>
     <string name="file_migration_dialog_title">Mise à jour du chemin de stockage</string>
     <string name="file_migration_directory_already_exists">Le dossier de données existe déjà. Choisissez une des options suivantes :</string>
-    <string name="file_migration_failed_dir_already_exists">Le dossier Nextcloud existe déjà</string>
+    <string name="file_migration_failed_dir_already_exists">Le dossier Infomaniak Drive existe déjà</string>
```
Comment (AndyScherzinger) [inline src/main/res/values-fr/strings.xml]: will be overridden by the bot, rather set a placeholder and replace it via code (that way the translations will come in and it'll work for your fork too!

File: src/main/res/values-es/strings.xml [hunk 2]
```diff
@@ -554,7 +552,7 @@
     <string name="screenshot_05_autoUpload_heading">Subida automática</string>
     <string name="screenshot_05_autoUpload_subline">para sus fotos &amp; vídeos</string>
     <string name="screenshot_06_davdroid_heading">Calendario y contactos</string>
-    <string name="screenshot_06_davdroid_subline">Sincronizar con DAVx5</string>
+    <string name="search_instruction">Buscar archivo (mínimo 2 caracteres)</string>
```
Comment (AndyScherzinger) [inline src/main/res/values-es/strings.xml]: will be overridden by the transifex bot

**Comment by Shagequi** [comment]
@AndyScherzinger 
Ok, I've fixed the translations (it was clearly a mistake, it shouldn't be present in this branch, sorry ^^') 

And i formated the code in the last commit (you may see code changes in previous commits)

File: src/main/res/values-es/strings.xml (hunk 2 above)
Comment (Shagequi) [reply src/main/res/values-es/strings.xml]: It's clearly a mistake, sorry

File: src/main/java/com/owncloud/android/ui/fragment/ExtendedListFragment.java [hunk 3]
```diff
@@ -135,7 +136,8 @@
 
     private EmptyRecyclerView mRecyclerView;
 
-    protected SearchView searchView;
+    public SearchView searchView;
+    public ImageView closeButton;
```
Comment (tobiasKaminsky) [inline src/main/java/com/owncloud/android/ui/fragment/ExtendedListFragment.java]: both are not needed to be public, according to AndroidStudio.
searchView can remain protected and closeButton even private

**Comment by tobiasKaminsky** [comment]
Tested & works :+1:

**Comment by Shagequi** [comment]
@tobiasKaminsky 

Thank you for your review.
There's no problem if you want to change my code, you know ^^

If these are impacting changes that require me to do a "heavy" review, I will gladly go over them :)
On the other hand, if it's a concern about the accessibility of a variable, or formatting, or obsolete import, don't hesitate to change things for me (we'll both save time).

**Pull request merged**